sympy/combinatorics/fp_groups.py

```diff
@@ -279,6 +285,54 @@ def __str__(self):
     __repr__ = __str__
 
 
+class FpSubgroup(DefaultPrinting):
+    '''
+    The class implementing a subgroup of an FpGroup or a FreeGroup
+    (only finite index subgroups are supported at this point). This
+    is to be used if one wishes to check if an element of the original
+    group belongs to the subgroup
+
+    '''
+    def __init__(self, G, gens):
+        super(FpSubgroup,self).__init__()
+        self.parent = G
+        self.generators = list(set([g for g in gens if g != G.identity]))
+        self.C = None
+
+    def __contains__(self, g):
+        gens = [s for s in g.contains_generators()]
+        if any([s not in self.generators for s in gens]):
+            return False
+        elif isinstance(self.parent, FreeGroup):
+            return True
```

Comment:
`self.generators` contains the generators of the subgroup. So that, say, if the parent group is a free group `F` on `a, b` and we create an `FpSubgroup(F, [a])`, `self.generators == [a]`. Then the first `if` checks if all of the letters of the word are in the subgroup alphabet, and if they are, then it belongs to the subgroup.

Though I've just realised that we've discussed this before and that there is a problem with it because the subgroup generators could be arbitrary words, like `a*b`. Sorry about that. Will need to think more carefully about how to check the membership for `FreeGroup` subgroups.